### ISSUE TITLE
Evaluate secret scanning tools

### ISSUE THREAD

**Issue by timoludwig** [digitalfabrik/integreat-cms#2250]
### Motivation

We use a few third-party tools which we need secrets for:
- DeepL
- SUMM.AI
- TextLab/HIX
- Firebase Cloud Messaging

In addition to the default secrets in our Django core base:
- SECRET_KEY
- DB_PASSWORD
- EMAIL_HOST_PASSWORD

We want to make sure we don't accidentally push such changes to the repository.

### Proposed Solution

Evaluate secret scanning tools out there, including e.g.:

- https://github.com/Yelp/detect-secrets
- https://github.com/GitGuardian/ggshield

and add pre-commit hooks for them.

### Alternatives

Rely on the fact that we recommend to put all secrets in the config file outside of the repository

**Comment by ulliholtgrave** [comment]
We are currently doing a benchmarking project of secret scanners at our research group. I guess we can use the results for our pipeline once they are public :)

**Comment by osmers** [comment]
@ulliholtgrave are the results public yet? I would keep the issue open for now and move it up the prioritization, hopefully your mentioned results can help.

**Comment by ulliholtgrave** [comment]
@osmers Yes, we do have results but due to all of the AI tools, they are already outdated and some of investigated scanners are also no longer maintained. I would recommend to take a look at the official GitHub actions to find a well-established tool and add it as an additional job in the pipeline.

**Comment by osmers** [comment]
Perfect - thank you :) I will ammend the scope/description of the research issue then to your suggestion!
